src/java/voldemort/store/InsufficientOperationalNodesException.java

```diff
@@ -35,56 +33,45 @@
 
     private static final long serialVersionUID = 1L;
 
-    private Collection<? extends Throwable> causes;
-
     private static final Logger logger = Logger.getLogger(InsufficientOperationalNodesException.class);
 
     public InsufficientOperationalNodesException(String s, Throwable e) {
         super(s, e);
-        causes = Collections.singleton(e);
     }
 
     public InsufficientOperationalNodesException(String s) {
         super(s);
-        causes = Collections.emptyList();
     }
 
     public InsufficientOperationalNodesException(Throwable e) {
         super(e);
-        causes = Collections.singleton(e);
     }
 
-    public InsufficientOperationalNodesException(Collection<? extends Throwable> failures) {
+    public InsufficientOperationalNodesException(List<? extends Throwable> failures) {
         this("Insufficient operational nodes to immediately satisfy request.", failures);
     }
 
-    public InsufficientOperationalNodesException(String message,
-                                                 Collection<? extends Throwable> failures) {
-        super(message, failures.size() > 0 ? failures.iterator().next() : null);
-        this.causes = failures;
+    public InsufficientOperationalNodesException(String message, List<? extends Throwable> failures) {
+        super(message, failures.size() > 0 ? failures.get(0) : null);
     }
 
     public InsufficientOperationalNodesException(String message,
                                                  List<Node> replicationSet,
                                                  List<Node> preferenceList,
                                                  List<Node> failedList,
-                                                 Collection<? extends Throwable> failures) {
+                                                 List<? extends Throwable> failures) {
         this(message + " Original replication set :" + stripNodeIds(replicationSet)
                      + " Known failed nodes before operation :"
                      + stripNodeIds(difference(replicationSet, preferenceList))
                      + " Estimated live nodes in preference list :" + stripNodeIds(preferenceList)
                      + " New failed nodes during operation :"
                      + stripNodeIds(difference(failedList, replicationSet)),
-             failures.size() > 0 ? failures.iterator().next() : null);
+             failures.size() > 0 ? failures.get(0) : null);
```

Comment:
More the code change, more possibility of regression. It could be written much better, but I don't want to fix it, unless there is a problem.